pkg/status/combinedstatus-resolution.go

```diff
@@ -1008,3 +1038,46 @@ func sortedStringSlice(s []string) []string {
 	sort.Sort(sort.StringSlice(s))
 	return s
 }
+
+// objectIsQueried checks whether `obj` appears in `query` as a standalone
+// occurrence.
+// That means, it must not be surrounded by alphanumeric characters.
+// query is assumed to be non-nil.
+func objectIsQueried(query *string, obj string) bool {
```

Comment:
Why does this take a _pointer to_ a string instead of simply a string?

Reply:
To make it slightly more efficient under the non-nil guarantee., assuming this will become a hot-function.

Passing a string by value copies 16 bytes, passing by pointer copies 8 bytes. I haven't ran benchmarks on whether this brings more value given the dereferences.